### ISSUE TITLE
Implement a CMO ID generation system + unit testing

### ISSUE THREAD

**Issue by ao508** [mskcc/cmo-metadb#56]
- Make a client to get these requests
- CMO_SAMPLE_ID_REQUEST as the topic (I changed label to ID - Ben)
- Need a consumer of this ^ request/reply that interfaces 
- for now also hardcode the patient id request that we would getting in the future from crdb (metadb should be checking if patient exists already in metadb or not, if not then go to crdb to get patient id)
- if we get the same request then do not generate the same id

during transition phase (to keep in mind for later):
- we will be getting the sample metadata from voyager, generate a cmo id and create the metadb node + the linked cmo sample node

** basically port allan's code, take what we get in the sample metadata and reply with generated cmo sample id 
https://github.mskcc.org/cmo/gen_cmo_tag

**Comment by ao508** [comment]
https://github.com/mskcc/LimsRest/blob/master/src/main/java/org/mskcc/limsrest/service/cmoinfo/SampleTypeCorrectedCmoSampleIdGenerator.java


You’re in the right spot. The generation happens mainly during sample promotion[*] but it can be triggered manually using a LIMS plugin as well.

https://github.com/mskcc/LimsRest/blob/36f484e5c4a9bb4397e01ebb8f28a8fc4e561327/src/main/java/org/mskcc/limsrest/service/PromoteBanked.java#L362
